lib/bpmachine/process_specification.rb

```diff
@@ -1,15 +1,13 @@
 module BPMachine
   module ProcessSpecification
-    def self.after_processes(&block)
-      after_process_actions << block
-    end
-
-    def self.after_process_actions
-      @after_process_actions ||= []
-    end
-
     def self.included(klass)
       klass.extend ClassMethods
+
+      klass.class_eval do
+        def self.included(klass)
+          klass.instance_variable_set :@bpm_module, self
```

Comment:
Yeah, I know. It's batshit ugly.
1. An `included` inside a `class_eval` inside another `included`?! WHAT?!
2. Who needs the `instance_variable_set`/`instance_variable_set` pair?